tt_metal/impl/dispatch/command_queue.hpp

```diff
@@ -501,22 +525,27 @@ class HWCommandQueue {
     friend void EnqueueTraceImpl(CommandQueue& cq);
     friend void EnqueueProgramImpl(CommandQueue& cq, std::variant < std::reference_wrapper<Program>, std::shared_ptr<Program> > program, bool blocking);
     friend void EnqueueReadBufferImpl(CommandQueue& cq, std::variant<std::reference_wrapper<Buffer>, std::shared_ptr<Buffer> > buffer, void* dst, bool blocking);
-    friend void EnqueueWriteBufferImpl(CommandQueue& cq, std::variant<std::reference_wrapper<Buffer>, std::shared_ptr<Buffer> > buffer, const void* src, bool blocking);
+    friend void EnqueueWriteBufferImpl(CommandQueue& cq, std::variant<std::reference_wrapper<Buffer>, std::shared_ptr<Buffer> > buffer, HostBufferMemTypes src, bool blocking);
```

Comment:
Why HostBufferMemTypes, isn't there already BufferType?

Edit: Now I see it's different, but maybe my initial confusion comes from the naming. Can you think of a better name?

Reply:
Renamed this to `HostDataType`.